lib/builder/builder.js

```diff
@@ -62,20 +62,31 @@ export default class Builder {
     // }
   }
 
+  pluginBasename(src) {
+    return path.basename(src, path.extname(src))
+      .replace(/[^a-zA-Z?\d\s:]/g, '')
+  }
+
   get plugins() {
+    let exists
     return _.uniqBy(
       this.options.plugins.map((p) => {
-        if (typeof p === 'string') p = { src: p }
-        const pluginBaseName = path.basename(p.src, path.extname(p.src)).replace(
-          /[^a-zA-Z?\d\s:]/g,
-          ''
-        )
-        return {
-          src: this.nuxt.resolveAlias(p.src),
-          ssr: p.ssr !== false,
-          name: 'nuxt_plugin_' + pluginBaseName + '_' + hash(p.src)
+        if (typeof p === 'string') {
+          p = { src: p }
+        }
+        p.basename = this.pluginBasename(p.src)
+        return p
+      }).filter((p) => {
+        exists = fs.existsSync(this.nuxt.resolveAlias(p.src))
+        if (!exists) {
+          consola.warn(`Plugin not found: ${p.basename}`)
```

Comment:
I think you can use reduce instead of filter+map.
And I didn’t see any pleace else in this method is using basename, so it should be unnecessary to extract it calculation